restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/client/api/CallApiClient.java

```diff
@@ -0,0 +1,163 @@
+    }
+
+    private void onDownloaderResponse(Object message, ActorRef self, ActorRef sender) {
+        final DownloaderResponse response = (DownloaderResponse) message;
+        if (logger.isInfoEnabled()) {
+            logger.info("Call api response succeeded " + response.succeeded());
+            logger.info("Call api response: " + response);
+        }
+        requestee.tell(message, self);
+    }
+
+    protected void onHangup(Hangup message, ActorRef self, ActorRef sender) throws URISyntaxException, ParseException {
+        requestee = sender;
+        callDetailRecord = getCallDetailRecord();
+        if(callDetailRecord == null){
+            logger.error("could not retrieve conferenceDetailRecord");
+        } else {
+            try {
+                URI uri = new URI("/restcomm"+callDetailRecord.getUri());
+                uri = UriUtils.resolve(uri);
+                if (logger.isInfoEnabled())
+                    logger.info("call api uri is: "+uri);
+
+                Header[] headers = {
+                        new BasicHeader(HttpHeaders.AUTHORIZATION, getSuperAdminAuthenticationHeader())
+                        ,new BasicHeader("Content-type", "application/x-www-form-urlencoded")
+                        ,new BasicHeader("Accept", "application/json")
+                    };
+
+                ArrayList<NameValuePair> postParameters = new ArrayList<NameValuePair>();
+                postParameters.add(new BasicNameValuePair("Status", "Completed"));
+
+                httpAsycClientHelper = httpAsycClientHelper();
+                HttpRequestDescriptor httpRequestDescriptor =new HttpRequestDescriptor(uri, "POST", postParameters, -1, headers);
+                httpAsycClientHelper.tell(httpRequestDescriptor, self);
+            } catch (Exception e) {
+                logger.error("Exception while trying to terminate call via api: ", e);
+            }
+        }
+    }
+
+    private String getSuperAdminAuthenticationHeader(){
+        Account superAdminAccount = storage.getAccountsDao().getAccount(SUPER_ADMIN_ACCOUNT_SID);
+
+        String auth = superAdminAccount.getSid() + ":" + superAdminAccount.getAuthToken();
+        byte[] encodedAuth = Base64.encodeBase64(auth.getBytes(Charset.forName("ISO-8859-1")));
+        return "Basic " + new String(encodedAuth);
+    }
+
+    private CallDetailRecord getCallDetailRecord() {
+        if(callSid != null){
+            return storage.getCallDetailRecordsDao().getCallDetailRecord(callSid);
+        }
+        return null;
+    }
+
+    private ActorRef httpAsycClientHelper(){
+        final Props props = new Props(new UntypedActorFactory() {
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public UntypedActor create() throws Exception {
+                return new HttpAsycClientHelper();
+            }
+        });
+        return getContext().actorOf(props);
+
+    }
+
+    @Override
+    public void postStop () {
```

Comment:
@maria-farooq as far as I can see, CallApiClient will never be destroyed and thus this method will never be called, see my comments on ConferencesEndpoint

Reply:
i have changed both CallApiClient and ConferenceApiClient to be disposable (used for one time request), they will clean them selves after sending back DownloaderResponse